### PR TITLE
Cpp executor - first collation conformance tests with ICU4C

### PR DESCRIPTION
**Pull request by sven-oly**
This adds files for compiling and executing C++ with ICU4C in conformance testing.  This is not yet ready for integration.

### CODE REVIEW

**Comment by sven-oly** [comment]
I think this needs to explicitly reference the particular JSON library.

Running ./executor gives with any input gives:
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string: construction from null is not valid

**Comment by sven-oly** [comment]
Yikes, some of my Schema stuff has leaked into this.

File: executors/cpp/Makefile [hunk 1]
```diff
@@ -8,6 +8,10 @@
 #  - configure, build, install ICU (make install)
 #  - make sure "icu-config" (in the ICU installed bin directory) is on
 #     the path
+# export ICU4C= $location
+# export LD_LIBRARY_PATH=$ICU4C/lib:$LD_LIBRARY_PATH
+# export PATH=$ICU4C/bin:$PATH
+# export CPATH=$ICU4C/bin:$CPATH
 #  - do 'make' in this directory
```
Comment (sffc) [inline executors/cpp/Makefile:16]: I don't see json-c mentioned anywhere; seems like it should be?

**Comment by sven-oly** [comment]
Updated this PR with ICU4C collation versions 71, 72, 73, and 74.

File: executors/cpp/coll.cpp [hunk 2]
```diff
@@ -60,7 +70,23 @@ const string test_collator(json_object *json_in)  //
...
+  } else {
+    has_locale = false;
+  }
+
+  bool ignore_punctuation = false;
+  json_object *ignore_obj = json_object_object_get(json_in, "ignorePunctuation");
+
+  if (ignore_obj) {
+    ignore_punctuation = true;
+    status = U_ZERO_ERROR;
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply executors/cpp/coll.cpp:85]: Removed status reset.

**Comment by echeran** [comment]
Many of the comments that you replied to don't have the changes applied in the PR. Did you make the commits locally without pushing them? Check Sublime Merge to visually see what's going on.

File: executors/cpp/coll.cpp [hunk 3]
```diff
@@ -70,57 +75,41 @@ const string test_collator(json_object *json_in)  //
...
+
+    // Include data compared in the failing test
+    json_object_object_add(
+        return_json, "s1", json_object_new_string(string1.c_str()));
+    json_object_object_add(
+        return_json, "s2", json_object_new_string(string2.c_str()));
+
+    // What was the actual returned value?
+    json_object_object_add(
+        return_json, "compare", json_object_new_int64(numeric_result));
```
Comment (echeran) [inline executors/cpp/coll.cpp:107]: yay, looking good (by using `numeric_result` as the arg to `json_object_new_int64`)

File: executors/test_strings.txt
Comment (echeran) [inline executors/test_strings.txt:1]: Thanks for updating this file accordingly.

**Review by echeran — APPROVED** [review]
LGTM!

**Pull request merged**